### ISSUE TITLE
changed_python_files_in_tree returns too many changed files

### ISSUE THREAD

**Issue by cfournie** [Shopify/shopify_python#53]
`shopify_python.git_utils.changed_python_files_in_tree` performs a diff between `active_branch` and `remotes/origin/master`. This causes too many files to be linted when used to check for added files if a local branch has not been rebased from `remotes/origin/master` recently (it includes files added to `remotes/origin/master` that haven't been included in `active_branch`).

**Comment by damnMeddlingKid** [comment]
I think all thats needed here is to change 

https://github.com/Shopify/shopify_python/blob/master/shopify_python/git_utils.py#L52

to 

```
modified = git_repo.git.diff("origin/{branch}..{branch}".format(active_branch), name_only=True, diff_filter="d")
```
i haven't tested the `diff_filter` arg but i looked at the library and it just passes it down to a shell invocation.

**Comment by JasonMWhite** [comment]
@damnMeddlingKid if I understand your invocation correctly, you're comparing `branch` to `origin/branch`, which would compare the difference your branch vs the pushed version of your branch. That's a bit different approach to what we've used in the past.

That reduces the scope of what needs to be linted significantly. On other repos, we also use this to determine what needs to be run through `mypy`, and I think this could open up cases where something would pass locally but fail globally, or worse, fail locally but pass globally.

I think I'd rather find the latest SHA that both `remote/master` and `active_branch` have in common, and diff from there. Effectively, lint all the files that have been added or modified on the current branch. That should be closer to what we were originally doing. What do you think?

**Comment by damnMeddlingKid** [comment]
Good point. I also dont know how this would behave on the first push.

**Comment by cfournie** [comment]
> I think I'd rather find the latest SHA that both remote/master and active_branch have in common, 

We want to find the [`merge-base`](https://git-scm.com/docs/git-merge-base) between `origin/master` and `active_branch` and diff that for changed files.

`origin/active_branch..active_branch` would diff too many files if the local `active_branch` was rebased from master but `origin/active_branch` was not.

**Comment by JasonMWhite** [comment]
Fixed by #54.
